python/ray/autoscaler/v2/scheduler.py

```diff
@@ -1024,9 +1024,17 @@ def _select_nodes_to_terminate(
         # Sort the nodes for termination.
         nodes.sort(key=ResourceDemandScheduler._sort_nodes_for_termination)
 
+        # Remove the head node from the list.
+        head_node = None
+        for i, node in enumerate(nodes):
+            if node.node_kind == NodeKind.HEAD:
+                # Remove the head node from the list.
+                head_node = nodes.pop(i)
+                break
+
         terminated_nodes, remained_nodes = (
             nodes[:num_to_terminate],
-            nodes[num_to_terminate:],
+            nodes[num_to_terminate:] + ([head_node] if head_node else []),
```

Comment:
can head_node be None?

Reply:
yes - commented, i.e. when the head node was just allocated, but not yet reported ray running.